src/components/Login/LoginContainer.test.tsx

```diff
@@ -0,0 +1,308 @@
+    mockRequestOTP.mockReset();
+    mockValidateOTP.mockReset();
+    cleanup();
+  });
+
+  it("should render LoginScanCard", () => {
+    expect.assertions(1);
+    const { getByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+  });
+
+  it("should render LoginMobileNumberCard if login endpoint + key in deep link", async () => {
+    expect.assertions(2);
+    mockDeepLink.mockImplementation(() =>
+      Promise.resolve({
+        queryParams: { key, endpoint },
+      })
+    );
+
+    const { getByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+    await waitFor(() => {
+      expect(getByTestId(loginMobileViewId)).not.toBeNull();
+    });
+  });
+
+  it("should render Id scanner", () => {
+    expect.assertions(3);
+    const { getByTestId, queryByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+    expect(getByTestId(barcodeScannerId)).not.toBeNull();
+    expect(queryByTestId(loadingViewId)).toBeNull();
+  });
+
+  it("valid QR detected and should render LoginMobileNumberCard", () => {
+    expect.assertions(4);
+    const { getByTestId, queryByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+    expect(getByTestId(barcodeScannerId)).not.toBeNull();
+    expect(queryByTestId(loadingViewId)).toBeNull();
+
+    fireEvent(getByTestId(barcodeScannerId), "onBarCodeScanned", {
+      nativeEvent: { data: `{"key": "${key}","endpoint": "${endpoint}"}` },
+    });
+    expect(getByTestId(loginMobileViewId)).not.toBeNull();
+  });
+
+  it("OTP requested successfully and should render LoginOTPCard", async () => {
+    expect.assertions(8);
+    mockRequestOTP.mockResolvedValueOnce({ status: "OK" });
+
+    const { getByTestId, queryByTestId, queryByText } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
+    expect(getByTestId(barcodeScannerId)).not.toBeNull();
+    expect(queryByTestId(loadingViewId)).toBeNull();
+
+    fireEvent(getByTestId(barcodeScannerId), "onBarCodeScanned", {
+      nativeEvent: { data: `{"key": "${key}","endpoint": "${endpoint}"}` },
+    });
+    expect(getByTestId(loginMobileViewId)).not.toBeNull();
```

Comment:
generally, do you think we need to have these repeated assertions that were part of previously written test cases?